dev-client/src/model/soilIdMatch/actions/soilIdMatchActions.ts

```diff
@@ -22,34 +22,57 @@ import {ThunkAPI} from 'terraso-client-shared/store/utils';
 import {Coords} from 'terraso-client-shared/types';
 
 import {
-  siteEntryForMatches,
+  ClientSoilIdFailureReason,
+  siteEntry,
   siteEntryForStatus,
   SoilIdEntry,
-  tempLocationEntryForMatches,
-  tempLocationEntryForStatus,
+  tempLocationEntry,
 } from 'terraso-mobile-client/model/soilIdMatch/soilIdMatches';
-import {AppState} from 'terraso-mobile-client/store';
+import {
+  updateSiteMatchesAfterTimeout,
+  updateTempMatchesAfterTimeout,
+} from 'terraso-mobile-client/model/soilIdMatch/soilIdMatchSlice';
+import {AppDispatch, AppState} from 'terraso-mobile-client/store';
+
+// TODO-cknipe: Should this live here or elsewhere? Also, change back to 20000
+const TIMEOUT_MS = 20000;
 
 export const fetchTempLocationBasedSoilMatchesThunk = async (
   coords: Coords,
   _: User | null,
-  __: ThunkAPI,
-) => fetchTempLocationBasedSoilMatches(coords);
+  {dispatch}: ThunkAPI,
+) => fetchTempLocationBasedSoilMatches(coords, dispatch);
+
+const timeoutError = {
+  __typename: 'SoilIdFailure' as const,
+  reason: 'TIMEOUT' as ClientSoilIdFailureReason,
+};
+type TimeoutError = typeof timeoutError;
+
+export type PromiseResult =
+  | Awaited<ReturnType<typeof soilIdService.fetchSoilMatches>>
+  | TimeoutError;
 
-export const fetchTempLocationBasedSoilMatches = async (coords: Coords) => {
-  const result = await soilIdService.fetchSoilMatches(coords, {
-    depthDependentData: [],
+export const fetchTempLocationBasedSoilMatches = async (
+  coords: Coords,
+  dispatch: AppDispatch,
+) => {
+  const timeoutPromise = new Promise<PromiseResult>(resolve => {
+    setTimeout(() => {
+      resolve(timeoutError);
+    }, TIMEOUT_MS);
   });
+  const apiPromise = soilIdService
+    .fetchSoilMatches(coords, {
+      depthDependentData: [],
+    })
+    .then(response => {
+      dispatch(updateTempMatchesAfterTimeout({coords, response}));
+      return response;
+    });
 
-  if (result.__typename === 'SoilIdFailure') {
-    return tempLocationEntryForStatus(coords, result.reason);
-  } else {
-    return tempLocationEntryForMatches(
-      coords,
-      result.matches,
-      result.dataRegion,
-    );
-  }
+  const result = await Promise.race([apiPromise, timeoutPromise]);
+  return tempLocationEntry(coords, result);
```

Comment:
question: can we encapsulate the promise race/timeout logic independently of the soil/API logic?